File: src/main/java/org/ventanas/Java/Ejercicio9.java
package org.ventanas.Java;
//9. Números primos
// Pedir un número al usuario y verificar si es un número primo.
import java.util.Scanner;

public class Ejercicio9 {
    public static void execute() {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Dame un número: ");
        try {
            int num = Integer.parseInt(scanner.nextLine());
            if (esPrimo(num)) {
                System.out.println(num + " es primo.");
            } else {
                System.out.println(num + " no es primo.");
            }
        } catch (Exception e) {
            System.out.println("Error al insertar el número, debe ser entero.");
        }
    }

    public static boolean esPrimo(int num) {
        if (num <= 1) {
            return false;
        }
        for (int i = 2; i <= Math.sqrt(num); i++) {
            if (num % i == 0) {
                return false;
            }
        }
        return true;
    }
}

File: src/main/java/org/ventanas/Kotlin/Ejercicio9.kt
package org.ventanas.Kotlin
//9. Números primos
// Pedir un número al usuario y verificar si es un número primo.

fun ejercicio9() {
    print("Dame un número: ")
    try {
        val num = readln().toInt()

        if (esPrimo(num)){
            println("$num es primo.")
        } else println("$num no es primo.")


    }catch (e:Exception){
        println("Error al insertar el numero, debe de ser entero.")
    }
}

fun esPrimo(num:Int): Boolean {
    var primo = num - 1
    var resultado = primo
    for (i in 2..<num){
        primo -= 1
        resultado *= primo

    }
    resultado += 1

    return resultado%num == 0 && num != 1
}